xapian-core/weight/lmweight.cc

```diff
@@ -159,8 +163,24 @@ LMWeight::get_sumpart(Xapian::termcount wdf, Xapian::termcount len,
 	weight_sum = (param_smoothing1 * weight_collection) +
 		     ((1 - param_smoothing1) * weight_document);
     } else if (select_smoothing == DIRICHLET_SMOOTHING) {
-	weight_sum = (wdf_double + (param_smoothing1 * weight_collection)) /
-		     (len_double + param_smoothing1);
+	// Enabling Dir+ weighting formula.
+	if (enable_dirplus) {
+
+	/* In the Dir+ weighting formula, sumpart weight contribution is :-
+	 *
+	 * sum of log of (1 + (wdf/(param_smoothing1 * weight_collection))) and 
+	 * log of (1 + (delta/param_smoothing1 * weight_collection))). 
+	 
+	 * Since, sum of logs iss log of product so weight_sum is calculated as product
+	 * of terms in log in the Dir+ formula.	
+	 */
```

Comment:
The comment should be aligned with the following code, not the containing block.